class-vip-support-api.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+class VIP_Support_REST_Controller {
+
+	private static $namespace;
+
+	function __construct() {
+		self::$namespace = 'vip/v1';
+		add_action( 'rest_api_init', [ $this, 'rest_api_init' ] );
+	}
+
+	function permission_callback() {
+		return wpcom_vip_go_rest_api_request_allowed( self::$namespace );
+	}
+
+	function rest_api_init() {
+		register_rest_route( self::$namespace, '/support-user', array(
+
+			// POST /vip/v1/support-user
+			array(
+				'methods' => WP_REST_Server::CREATABLE,
+				'permission_callback' => array( $this, 'permission_callback' ),
+				'args' => array(
+					'user_login' => array(
+						'required' => true,
+					),
+					'user_pass' => array(
+						'required' => true,
+						'validate_callback' => function( $param, $request, $key ) {
+							if ( strlen( $param ) < 10 ) {
```

Comment:
`-data 'user_pass=aaaaaaaaaa'` <= super secure ;)

(No change needed here; this is fine)